packages/cloudflare/src/cli/build/build-next-app.ts

```diff
@@ -7,17 +8,15 @@ import { execSync } from "node:child_process";
  *
  * @param nextAppDir the directory of the app to build
  */
-export function buildNextjsApp(nextAppDir: string): void {
-  runNextBuildCommand("pnpm", nextAppDir);
+export async function buildNextjsApp(nextAppDir: string): Promise<void> {
+  const pm = await detect();
+
+  runNextBuildCommand(pm?.name ?? "npm", nextAppDir);
 }
 
 // equivalent to: https://github.com/sst/open-next/blob/f61b0e94/packages/open-next/src/build.ts#L175-L186
-function runNextBuildCommand(
-  // let's keep things simple and just support only pnpm for now
-  packager: "pnpm" /*"npm" | "yarn" | "pnpm" | "bun"*/,
-  nextAppDir: string
-) {
-  const command = ["bun", "npm"].includes(packager) ? `${packager} next build` : `${packager} next build`;
+function runNextBuildCommand(packager: "npm" | "pnpm" | "yarn" | "bun", nextAppDir: string) {
```

Comment:
nit: use [`AgentName`](https://github.com/antfu-collective/package-manager-detector/blob/5c494d7a2fcad6a900d0a666bca5601ade003aa9/src/types.ts#L53) for the type

Reply:
yeah I considered that, but I dislike that type name 😅 

would be ok if I included it but rename it?
as in:
```
type PM = AgentName;
```

Reply:
I've ended up with `import { type AgentName as PackageManager } from "package-manager-detector";`

Hopefully this works for you 😄